delta_backend/src/ConversionChecker.py

```diff
@@ -85,25 +93,121 @@ def convertData(self, expressionType, expressionRule, fieldName, fieldValue):
+            # Accept "YYYY-MM-DD" and return as is
+            if re.match(r"^\d{4}-\d{2}-\d{2}$", fieldValue):
+                try:
+                    dt = datetime.strptime(fieldValue, "%Y-%m-%d")
+                    if dt.date() > datetime.now(ZoneInfo("UTC")).date():
+                        if report_unexpected_exception:
+                            self._log_error(fieldName, fieldValue, "Date cannot be in the future")
+                        return ""
+                    return fieldValue
+                except ValueError:
+                    if report_unexpected_exception:
+                        self._log_error(fieldName, fieldValue, "Invalid date format")
+                    return ""
+            try: 
+                # Parse ISO format with or without microseconds and TZ
+                dt = datetime.fromisoformat(fieldValue)
+            except ValueError: 
+                if report_unexpected_exception:
+                    self._log_error(fieldName, fieldValue, "Invalid date format")
+                return ""
+
+            # Assign UTC if tzinfo is missing
+            if dt.tzinfo is None:
+                dt = dt.replace(tzinfo=ZoneInfo("UTC"))
+            
+            now_utc = datetime.now(ZoneInfo("UTC"))
+            if dt.astimezone(ZoneInfo("UTC")) > now_utc:
+                if report_unexpected_exception:
+                    self._log_error(fieldName, fieldValue, "Date cannot be in the future")
+                return ""
+
+            # Validate timezone offset
+            offset = dt.utcoffset()
+            allowed_offsets = [
+                ZoneInfo("UTC").utcoffset(dt),
+                ZoneInfo("Europe/London").utcoffset(dt),
+            ]
+
+            if offset not in allowed_offsets:
+                if report_unexpected_exception:
+                    self._log_error(fieldName, fieldValue, f"Unsupported offset: {offset}")
+                return ""
+
+            dt_utc = dt.astimezone(ZoneInfo("UTC")).replace(microsecond=0)
+
+            # Format and return with custom suffix
+            formatted = dt_utc.strftime("%Y%m%dT%H%M%S%z")
+            return formatted.replace("+0000", "00").replace("+0100", "01")
+        
+        # For all other fields, apply standard %Y%m%d processing
+        if format_str == "%Y%m%d":
+            fieldValue = fieldValue.replace("-", "").replace("/", "")
+            # Validate expected raw input format if using %Y%m%d
+            if not re.match(r"^\d{8}$", fieldValue):
+                if report_unexpected_exception:
+                    self._log_error(fieldName, fieldValue, "Date must be in YYYYMMDD format")
+                return ""
+
         try:
-            dt = datetime.fromisoformat(fieldValue)
-            format_str = expressionRule.replace("format:", "")
+            dt = datetime.strptime(fieldValue, format_str)
+
+            # Reject future dates if the field is BirthDate
+            if fieldName in "contained|#:Patient|birthDate":
```

Comment:
I don't love referring to field names in the conversion logic - can this be controlled by the schema instead? Same question for `recorded` above